frontend/dockerfile/instructions/commands_runmount.go

```diff
@@ -241,10 +233,11 @@ func parseMount(value string, expander SingleWordExpander) (*Mount, error) {
 		case "id":
 			m.CacheID = value
 		case "sharing":
-			if _, ok := allowedSharingTypes[strings.ToLower(value)]; !ok {
-				return nil, errors.Errorf("unsupported sharing value %q", value)
+			v := ShareMode(strings.ToLower(value))
+			if _, ok := allowedSharingModes[v]; !ok {
+				return nil, suggest.WrapError(errors.Errorf("unsupported sharing value %q", value), value, allShareModes(), true)
```

Comment:
I based this on df9781b46c4ed4f4e7146c2241c0cb513167c2c5, but am slightly confused, as the `suggest.WrapError()` uses `caseSensitive=true`, but the code here suggests otherwise (perhaps this was meant to be an optimisation to not have to do the `strings.ToLower()`? But in that case we should pass `v` to the function, not `value`.